src/ui/public/field_wildcard/field_wildcard.js

```diff
@@ -0,0 +1,35 @@
+import { escapeRegExp, memoize } from 'lodash';
+
+export default function fieldWildcard(config) {
+  const metaFields = config.get('metaFields');
+
+  const makeRegEx = memoize(function makeRegEx(glob) {
+    return new RegExp('^' + glob.split('*').map(escapeRegExp).join('.*') + '$');
+  });
+
+  function endsWith(str, test) {
+    if (!str || typeof str !== 'string' || str.length < test.length) {
+      return false;
+    }
+    return str.substring(str.length - test.length) === test;
+  }
+
+  function fieldWildcardMatcher(globs) {
+    return function matcher(val) {
+      // do not test metaFields or keyword
+      if (metaFields.indexOf(val) !== -1 || endsWith(val, '.keyword')) {
```

Comment:
You could use Lodash's endsWith instead: https://lodash.com/docs/3.10.1#endsWith